### DISCUSSION TITLE
Communicate between frontend (browser-electron)and backend(node-electron) throw IPC

### DISCUSSION THREAD

**Discussion by YOSSTM** [eclipse-theia/theia#10776] [Q&A] [answered]
I have a struggle how to communicate through IPC messages between the browser-electron and the node-electron in theia context.
Can I find any example using theia dealing with that topic ?
Thanks

**Comment by msujew** [comment, marked as answer]
Hi @YOSSTM,

do you mean communication between the frontend and the server backend (browser -> node) or the frontend and the electron main process (browser -> main)? I'm asking since we have a `electron-node` folder structure which is running electron related code in the server backend process and a `electron-main` folder, which contains the main process code.

Given that you try to use the IPC messaging system, I assume that you want browser -> main communication. I would simply recommend you to use the existing proxy infrastructure, which you can see bound in the [frontend](https://github.com/eclipse-theia/theia/blob/cdc2d9f21a6edf82050882be5f8a6538e436d053/packages/core/src/electron-browser/window/electron-window-module.ts#L28-L30) and [backend](https://github.com/eclipse-theia/theia/blob/cdc2d9f21a6edf82050882be5f8a6538e436d053/packages/core/src/electron-main/electron-main-application-module.ts#L46-L50) for the `ElectronMainWindowService`.

If you are actually communicating between the frontend and the server backend, you should use the existing websocket proxy, like you would do in a browser Theia application.

**Reply by YOSSTM** [reply]
@msujew No I am trying to communicate from the electron main process node-electron(backend for the electron) to the electron-browser (browser for the electron), but still not found a documentation to do so and how to handle that .Just what I found is a way to bind a backend service contribution to frontend using jsonrpc communication .Thanks

**Reply by msujew** [reply]
I see, yeah there isn't any special documentation on it, since that's a rare use case for most users. But like I've said, you can use the proxy infrastructure in my two links to use the IPC communication to call a service implementation.
